File: pinyougou-web-user/src/main/webapp/js/controller/addressController.js
//首页控制器
app.controller('addressController',function($scope,loginService,addressService){
    //获取当前用户的地址列表
    $scope.findAddressList=function(){
        addressService.findAddressList().success(
            function(response){
                $scope.addressList=response;
                /*                for(var i=0;i<$scope.addressList.length;i++){
                                    if($scope.addressList[i].isDefault=='1'){
                                        $scope.address=$scope.addressList[i];
                                        break;
                                    }
                                }*/

            }
        );
    }
    //获取当前用户的地址列表
    $scope.deleAddr = function(id){
        addressService.deleAddr(id).success(function(response){
            // 判断保存是否成功:
            if(response.flag == true){
                alert(response.message)
                $scope.findAddressList();
            }else{
                // 删除失败
                alert(response.message);
            }
        });
    }
    //省
    $scope.findAllAddress = function(){
            addressService.findAllAddress().success(function(response){
                $scope.provinces = response;
            });
    }
    // 市
    $scope.$watch("entity.provinceid",function(newValue,oldValue){
        addressService.findCityById(newValue).success(function(response){
            $scope.cities = response;
        });
    });
    //  县
    $scope.$watch("entity.cityid",function(newValue,oldValue){
        addressService.findAreaById(newValue).success(function(response){
            $scope.areas = response;
        });
    });

    $scope.save = function () {
        addressService.save($scope.entity).success(function(response){
            if (response.flag = true ){
                alert(response.message)
                //重新查一下数据库刷新页面
                $scope.findAddressList();
            }else {
                alert(response.message)
            }
        })
    }

    $scope.showName=function(){
        loginService.showName().success(
            function(response){
                $scope.loginName=response.loginName;
            }
        );
    }

});

File: pinyougou-web-user/src/main/java/cn/itcast/core/controller/AddressController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.address.*;
import cn.itcast.core.service.AddressService;
import com.alibaba.dubbo.config.annotation.Reference;
import com.sun.org.apache.regexp.internal.RE;
import entity.Result;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 收货地址
 */
@RestController
@RequestMapping("/address")
public class AddressController {

    @Reference
    private AddressService addressService;

    //查询当前登陆人所有收货地址
    @RequestMapping("/findListByLoginUser")
    public List<Address> findListByLoginUser(){
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        return addressService.findListByLoginUser(name);
    }

    //删除用户地址
    @RequestMapping("/deleAddr")
    public Result deleAddr(Long id){
        try {
            addressService.deleAddr(id);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    //省
    @RequestMapping("/findAllAddress")
    public List<Provinces> findAllAddress(){
        List<Provinces> allAddress = addressService.findAllAddress();
        return allAddress;
    }
    //市
    @RequestMapping("/findCityById")
    public List<Cities> findCityById(String provinceid){
        return addressService.findCityById(provinceid);
    }
    //县
    @RequestMapping("/findAreaById")
    public List<Areas> findAreaById(String cityid){
        return addressService.findAreaById(cityid);
    }

    @RequestMapping("/add")
    public Result save(@RequestBody Address entity){
        try {
            String name = SecurityContextHolder.getContext().getAuthentication().getName();
            addressService.save(entity,name);
            return new Result(true,"保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"保存失败");
        }
    }
}
